flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/stream/StreamExecMultiJoin.java

```diff
@@ -0,0 +1,444 @@
+    private static final String FIELD_NAME_CONFIGURATION = "configuration";
+    private static final String FIELD_NAME_INPUT_PROPERTIES = "inputProperties";
+    private static final String FIELD_NAME_OUTPUT_TYPE = "outputType";
+    private static final String FIELD_NAME_DESCRIPTION = "description";
+    private static final String FIELD_NAME_JOIN_TYPES = "joinTypes";
+    private static final String FIELD_NAME_JOIN_CONDITIONS = "joinConditions";
+    private static final String FIELD_NAME_GENERATED_MULTI_JOIN_CONDITION =
+            "generatedMultiJoinCondition";
+    private static final String FIELD_NAME_JOIN_ATTRIBUTE_MAP = "joinAttributeMap";
+    private static final String FIELD_NAME_INPUT_SIDE_SPECS = "inputSideSpecs";
+    private static final String FIELD_NAME_STATE_METADATA_LIST = "stateMetadataList";
+    private static final String FIELD_NAME_KEY_SELECTORS = "keySelectorsForTransformation";
+    private static final String FIELD_NAME_COMMON_KEY_TYPE = "commonKeyType";
+
+    @JsonProperty(FIELD_NAME_JOIN_TYPES)
+    private final List<JoinType> joinTypes;
+
+    @JsonProperty(FIELD_NAME_JOIN_CONDITIONS)
+    private final List<? extends @Nullable RexNode> joinConditions;
+
+    @JsonProperty(FIELD_NAME_JOIN_ATTRIBUTE_MAP)
+    private final Map<
+                    Integer,
+                    Map<
+                            AttributeBasedJoinKeyExtractor.AttributeRef,
+                            AttributeBasedJoinKeyExtractor.AttributeRef>>
+            joinAttributeMap;
+
+    @JsonProperty(FIELD_NAME_INPUT_SIDE_SPECS)
+    private final List<JoinInputSideSpec> inputSideSpecs;
+
+    @JsonProperty(FIELD_NAME_KEY_SELECTORS)
+    private final List<KeySelector<RowData, RowData>> commonJoinKeySelectorsForTransformation;
+
+    @JsonProperty(FIELD_NAME_COMMON_KEY_TYPE)
+    private final InternalTypeInfo<RowData> commonJoinKeyType;
+
+    @JsonProperty(FIELD_NAME_STATE_METADATA_LIST)
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    private final List<StateMetadata> stateMetadataList;
+
+    public StreamExecMultiJoin(
+            ReadableConfig tableConfig,
+            List<JoinType> joinTypes,
+            List<? extends @Nullable RexNode> joinConditions,
+            @Nullable RexNode generatedMultiJoinCondition,
+            Map<
+                            Integer,
+                            Map<
+                                    AttributeBasedJoinKeyExtractor.AttributeRef,
+                                    AttributeBasedJoinKeyExtractor.AttributeRef>>
+                    joinAttributeMap,
+            List<JoinInputSideSpec> inputSideSpecs,
+            Map<Integer, Long> stateTtlFromHint,
+            List<InputProperty> inputProperties,
+            RowType outputType,
+            String description,
+            List<KeySelector<RowData, RowData>> commonJoinKeySelectorsForTransformation,
+            InternalTypeInfo<RowData> commonJoinKeyType) {
```

Comment:
Same comment as in the KeySelector. Creating type information comes later in the translation process i.e. `translateToPlan`

Reply:
done